model-api/build.gradle.kts

```
@@ -58,6 +59,25 @@ kotlin {
     }
 }
 
+val generateVersionVariable by tasks.creating {
```

```suggestion
val generateVersionVariable by tasks.registering {
```
Prefer using `registering`, so that the [configuration can be deferred](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html).